client.go

```diff
@@ -10,31 +11,78 @@ import (
 	nebula "github.com/vesoft-inc/nebula-go/v2"
 )
 
-type Data []string
+type (
+	// NebulaPool nebula connection pool
+	NebulaPool struct {
+		HostList          []nebula.HostAddress
+		Pool              *nebula.ConnectionPool
+		Log               nebula.Logger
+		DataChs           []chan Data
+		OutoptCh          chan []string
+		Version           string
+		csvStrategy       csvReaderStrategy
+		initialized       bool
+		sessions          []*nebula.Session
+		channelBufferSize int
+		sslconfig         *sslConfig
+		mutex             sync.Mutex
+	}
 
-type Output struct {
-	TimeStamp    int64
-	NGQL         string
-	Latency      int64
-	ResponseTime int32
-	IsSucceed    bool
-	Rows         int32
-	ErrorMsg     string
-}
+	// NebulaSession a wrapper for nebula session, could read data from DataCh
+	NebulaSession struct {
+		Session *nebula.Session
+		Pool    *NebulaPool
+		DataCh  chan Data
+	}
+
+	// Response a wrapper for nebula resultset
+	Response struct {
+		*nebula.ResultSet
+		ResponseTime int32
+	}
+
+	csvReaderStrategy int
+
+	sslConfig struct {
+		rootCAPath     string
+		certPath       string
+		privateKeyPath string
+	}
+
+	// Data data in csv file
+	Data []string
+
+	output struct {
+		timeStamp    int64
+		nGQL         string
+		latency      int64
+		responseTime int32
+		isSucceed    bool
+		rows         int32
+		errorMsg     string
+	}
```

Comment:
no need errorCode?

Reply:
error code is less useful for user.

just like this
![image](https://user-images.githubusercontent.com/1726587/147550715-de47df73-ab3e-4718-abe6-3dc85a3b025a.png)